docs/ios/6x/api-reference/embrace-client.md

```diff
@@ -53,167 +53,335 @@ func start() throws -> Embrace
+static func recordSpan<T>(
+    name: String,
+    parent: Span? = nil,
+    type: SpanType = .performance,
+    attributes: [String: String] = [:],
+    block: (Span?) throws -> T
+) rethrows -> T
 ```
 
 **Parameters**:
 - `name`: The name of the span.
-- `type`: The type of the span, one of `SpanType` values.
-- `attributes`: Optional attributes to attach to the span.
+- `parent`: The parent `Span`, if this span is a child.
+- `type`: The type of the span.
+- `attributes`: A dictionary of attributes to set on the span.
+- `block`: The block to execute, receives a `Span` as an argument.
 
-**Returns**: A `Span` object representing the started span, or `nil` if the span could not be created.
+**Returns**: The result of the block.
 
-### User Identification
+**Note**: This static method validates the presence of the Embrace client and will call the block with a nil span if the client is not present, ensuring the block always executes.
 
-#### `setUserIdentifier(_:)`
+#### `recordCompletedSpan(...)`
 
-Sets the user identifier for the current session.
+Records a span after the fact with all details.
 
 ```swift
-func setUserIdentifier(_ identifier: String)
+func recordCompletedSpan(
+    name: String,
+    type: SpanType,
+    parent: Span?,
+    startTime: Date,
+    endTime: Date,
+    attributes: [String: String],
+    events: [RecordingSpanEvent],
+    errorCode: SpanErrorCode?
+)
 ```
 
-**Parameters**:
-- `identifier`: A unique identifier for the user.
+#### `flush(_:)`
 
-#### `setUserEmail(_:)`
-
-Sets the user's email address for the current session.
+Flushes a span to disk, useful for long-running spans.
 
 ```swift
-func setUserEmail(_ email: String)
+func flush(_ span: Span)
 ```
 
 **Parameters**:
-- `email`: The user's email address.
-
-#### `setUsername(_:)`
+- `span`: A `Span` object that implements `ReadableSpan`.
 
-Sets the username for the current session.
+#### Adding Events to Spans
 
 ```swift
-func setUsername(_ username: String)
+// Add single event to current session span
+func add(event: SpanEvent)
+
+// Add multiple events to current session span  
+func add(events: [SpanEvent])
 ```
 
-**Parameters**:
-- `username`: The username to set.
+### User Identification and Metadata
 
-#### `clearUserInfo()`
+User identification and session properties are managed through the `metadata` property, which provides access to a `MetadataHandler` instance.
 
-Clears all user information that has been set.
+#### Accessing the Metadata Handler
 
 ```swift
-func clearUserInfo()
+let metadata = Embrace.client?.metadata
 ```
 
-### Custom Properties
+#### User Properties
+
+Set user information using the metadata handler's properties:
+
+```swift
+// Set user identifier
+Embrace.client?.metadata.userIdentifier = "user123"
+
+// Set user email
+Embrace.client?.metadata.userEmail = "user@example.com"
+
+// Set username
+Embrace.client?.metadata.userName = "john_doe"
```

Comment:
we don't want people adding user name and email, and they should be deprecated from the metadata API on the SDK side. there's an eng task to do this eventually, but let's make sure not to document it here.

for userIdentifier, can you note that this should be anonymized ID that doesn't represent account info for the app user?

[more info here](url) if you want to see the latest time we've removed this feature